### PR TITLE
docs: fix outdated links to docs.docker.com

### PR DESCRIPTION
**Pull request by dvdksn**
Some of the updated links had redirects in place,
Updating them to current paths to avoid relying on old redirects.
This is not an exhaustive update to all links,
just a few links that I happened to pick up on.

my hacky reproducer:

```bash
#!/usr/bin/env bash

volume_old_re='/engine/tutorials/dockervolumes/#adding-a-data-volume' 
bind_old_re='/engine/tutorials/dockervolumes/#mount-a-host-(directory|file)-as-a-data-volume'
dockerfile_old_re='(/engine)?/reference/builder/'

volume_files=$(rg $volume_old_re -l -g 'content.md')
bind_files=$(rg $bind_old_re -l -g 'content.md')
dockerfile_files=$(rg $dockerfile_old_re -l -g 'content.md')

for file in $volume_files; do
	sed -i "s,${volume_old_re},/storage/volumes/,g" $file;
done

for file in $bind_files; do
	sed -i -r "s,${bind_old_re},/storage/bind-mounts/,g" $file;
done

for file in $dockerfile_files; do
	sed -i -r "s,${dockerfile_old_re},/reference/dockerfile/,g" $file;
done
```

Signed-off-by: David Karlsson <35727626+dvdksn@users.noreply.github.com>

### CODE REVIEW

**Review by whalelines — APPROVED** [review]
LGTM

**Review by thaJeztah — COMMENTED** [review]
LGTM

File: mariadb/content.md [hunk 1]
```diff
@@ -184,16 +184,16 @@ $ docker run --name some-mysql -e MARIADB_ROOT_PASSWORD_FILE=/run/secrets/mariad
...
+When a container is started for the first time, a new database with the specified name will be created and initialized with the provided configuration variables. Furthermore, it will execute files with extensions `.sh`, `.sql`, `.sql.gz`, `.sql.xz` and `.sql.zst` that are found in `/docker-entrypoint-initdb.d`. Files will be executed in alphabetical order. `.sh` files without file execute permission are sourced rather than executed. You can easily populate your `%%IMAGE%%` services by [mounting a SQL dump into that directory](https://docs.docker.com/engine/tutorials/dockervolumes/#mount-a-host-file-as-a-data-volume) and provide [custom images](https://docs.docker.com/reference/dockerfile/) with contributed data. SQL files will be imported by default to the database specified by the `MARIADB_DATABASE` variable.
```
Comment (yosifkit) [inline mariadb/content.md:187]: Oh, I see another if you want to update it as well: `https://docs.docker.com/engine/tutorials/dockervolumes/#mount-a-host-file-as-a-data-volume` (here and in `mysql` and `percona`).

(your script only got `https://docs.docker.com/engine/tutorials/dockervolumes/#mount-a-host-directory-as-a-data-volume`)

Comment (dvdksn) [reply mariadb/content.md]: Thanks - fixed these occurrences too!

**Review by yosifkit — APPROVED** [review]

**Review by tianon — APPROVED** [review]

**Pull request merged**